core/src/main/java/com/linecorp/armeria/server/throttling/AbstractThrottlingServiceBuilder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.throttling;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.server.Service;
+
+/**
+ * Builds a new {@link AbstractThrottlingService}.
+ * @param <I> type of the request
+ * @param <O> type of the response
+ */
+abstract class AbstractThrottlingServiceBuilder<I extends Request, O extends Response> {
+
+    @Nullable
+    private ThrottlingStrategy<I> strategy;
+    private ThrottlingAcceptHandler<I, O> acceptHandler;
```

Comment:
I think instead of having `default*Handler` methods we can initialize these to defaults instead of `null`. Doesn't hurt to add the accept handler to the constructor to be called with `super` in child builder.

Reply:
Kept the way it is

Reply:
Why?

Reply:
Calling super with the default handlers sounds better to me.

Reply:
> Why?
Because this is my design choice.
Defaults are implementation specific and those aren't parameters (which they would become when passing them in the constructor). A descendant is capable of defining its own defaults, and in case of "accept" it's obligated to do so.
Let me know if it violates any rules.

Reply:
While there are some concepts defined as rules in our contributing guidelines, many are just principles and we use code review to share to contributors. In this case, we use methods as an indicator of our user API and prefer not to use methods for passing parameters to a base class (I'll have to disagree that default behavior isn't a "parameter"). These default handlers should be constants, and there isn't much value in having all subclasses manage the boilerplate to make them constants. Every project will have its own style and level of rigor and that's what we have code review for.

Reply:
Defaults are defining the behavior of the each implementation - `ThrottlingServiceBuilder` and `ThrottlingRpcServiceBuilder`. Such behavior defined by the base class - `AbstractThrottlingServiceBuilder`. `defaultRejectHandler()` method is not part of the class hierarchy public API. And no one suggest/forces to create boilerplate anywhere.
Inheritance and this kind of non-public behavior enforcement used widely in Java programming, including Armeria. You can take `com.linecorp.armeria.client.retry.AbstractBackoff` hierarchy and its `doNextDelayMillis()` method.

Reply:
Fixed